src/pykeen/quadruples/quadruples_factory.py

```diff
@@ -0,0 +1,1246 @@
+# -*- coding: utf-8 -*-
+
+"""Implementation of basic instance factory which creates just instances based on standard tKG quadruples."""
+
+import logging
+import pathlib
+import warnings
+from datetime import datetime
+from typing import (
+    Any,
+    Collection,
+    Dict,
+    Iterable,
+    List,
+    Mapping,
+    MutableMapping,
+    Optional,
+    Sequence,
+    Set,
+    TextIO,
+    Union,
+    cast,
+)
+
+import numpy as np
+import pandas as pd
+import torch
+from class_resolver import HintOrType, OptionalKwargs
+from pykeen.instances import BatchedSLCWAInstances, SubGraphSLCWAInstances
+from pykeen.inverse import relation_inverter_resolver
+from pykeen.sampling import NegativeSampler
+from pykeen.triples.splitting import split
+from pykeen.triples.triples_factory import (
+    KGInfo,
+    Labeling,
+    _ensure_ids,
+    compact_mapping,
+    create_entity_mapping,
+    create_relation_mapping,
+    normalize_path,
+)
+from pykeen.triples.utils import tensor_to_df
+from pykeen.typing import (
+    LABEL_HEAD,
+    LABEL_RELATION,
+    LABEL_TAIL,
+    EntityMapping,
+    RelationMapping,
+    TorchRandomHint,
+)
+from pykeen.utils import format_relative_comparison
+from torch.utils.data import Dataset
+
+from .instances import LCWAQuadrupleInstances, SLCWAQuadrupleInstances
+from ..constants import COLUMN_TEMPORAL_LABELS
+from ..typing import LabeledQuadruples, MappedQuadruples, TimestampMapping
+
+__all__ = [
+    "QuadruplesFactory",
+    "INVERSE_SUFFIX",
+]
+
+logger = logging.getLogger(__name__)
+
+INVERSE_SUFFIX = "_inverse"
+
+QUADRUPLES_DF_COLUMNS = (
+    "head_id",
+    "head_label",
+    "relation_id",
+    "relation_label",
+    "tail_id",
+    "tail_label",
+    "timestamp_id",
+    "timestamp_label",
+)
+
+LABEL_TIMESTAMP = "timestamp"
+
+
+def load_quadruples(
+    path: Union[str, pathlib.Path, TextIO],
+    delimiter: str = "\t",
+    encoding: Optional[str] = None,
+    column_remapping: Optional[Sequence[int]] = None,
+) -> LabeledQuadruples:
+    """Load quadruples from original datafile."""
+    if encoding is None:
+        encoding = "utf-8"
+
+    if column_remapping is not None:
+        if len(column_remapping) != 4:
+            raise ValueError("remapping of temporal KG datasets must have length of 4")
+    df = pd.read_csv(
+        path, sep=delimiter, encoding=encoding, dtype=str, header=None, usecols=column_remapping
+    )
+    if column_remapping is not None:
+        df = df[[df.columns[c] for c in column_remapping]]
+    return df.to_numpy()
+
+
+def _get_quadruple_mask(
```

Comment:
I do not see any difference to the method used for triples: https://github.com/pykeen/pykeen/blob/746cbe0c963799a7e8d0597cdd5d35ec5d90816b/src/pykeen/triples/triples_factory.py#L147